### PR TITLE
Link component

### PR DESCRIPTION
**Pull request by NishantSinghhhhh**
Description

Component Name : Link component present in Sistent component folder ,

This PR addresses various enhancements and features, including:

Documentation : I have written Documentation for the Link Component Used in Sistent Repository
Link Variants: Added different styles for links, including underlined and colored versions to enhance navigation and visibility.

Security Practices: Documented best practices for using links with target="_blank", highlighting the importance of rel="noopener" and rel="noreferrer" attributes to improve security.

Notes for Reviewers

Please pay special attention to the documentation and security notes for external links. Feedback on these changes would be greatly appreciated.

[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

 Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 3465ed85bf57879e4cbb8922748f9615ebb8df50 at: https://6716033edc3fa002dc000554--layer5.netlify.app

**Comment by vishalvivekm** [comment]
@NishantSinghhhhh I added the item to the docs: 
https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?tab=t.0

**Comment by NishantSinghhhhh** [comment]
@vishalvivekm  thank you so much Sir

**Comment by l5io** [comment]
🚀 Preview for commit b0d80fde3c9494e9e24b41730864b2d52b9f0afe at: https://67167a8c43c3d76be6961631--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit e994890d9fc547c41bf5332fff001c9c0b029022 at: https://671681a0c892347e07f863a7--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 1dbe06a9e8b1afdb3730eadbad57f111d91cea78 at: https://67168a6c907aac827f285595--layer5.netlify.app

**Comment by NishantSinghhhhh** [comment]
I have implemented the necessary changes as requested in the pull request [#6015] Please review the updates, and if everything meets the project requirements, I would appreciate your help in merging the PR

File: src/sections/Projects/Sistent/components/link/index.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+          <p>
+          Basic anchor links with customizable colors and typography styles that inherit properties from the Typography component. This allows flexibility in styling with options like various text sizes or variants.
+          </p>
+          <Row $Hcenter className="image-container">
+            <SistentThemeProvider initialMode={isDark ? "dark" : "light"}>
+              <Link
+                href="https://layer5.io/"
+                style={{
+                  marginLeft: "10px",
+                  color: "green",
```
Comment (sudhanshutech) [inline src/sections/Projects/Sistent/components/link/index.js:77]: @NishantSinghhhhh don't pass any custom color, the link already has that required color which is the brand color.

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

**Review by sudhanshutech — CHANGES_REQUESTED** [review]
❌
{7E727F6D-B416-4DD6-A672-4A3F45D89E41}

✅
{1528FE11-9AFF-4F06-A020-315E56699D67}

File: src/sections/Projects/Sistent/components/link/index.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+const SistentLink = () => {
+  const location = useLocation();
+  const { isDark } = useStyledDarkMode();
+
+  return (
+    <SistentLayout title="Link" >
+      <div className="content">
+        <a id="Identity">
+          <h2>link</h2>
```
Comment (sudhanshutech) [inline src/sections/Projects/Sistent/components/link/index.js:19]: Uppercase please

File: src/sections/Projects/Sistent/components/link/guidance.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          </p>
+          <Row $Hcenter className="image-container">
+            <SistentThemeProvider initialMode={isDark ? "dark" : "light"}>
+              <Link
+                href="https://layer5.io/projects/sistent/components/link/guidance"
+                target="_blank"
+                rel="noopener noreferrer"
+                style={{
+                  marginLeft: "10px",
+                  textDecoration: "underline",
```
Comment (sudhanshutech) [inline src/sections/Projects/Sistent/components/link/guidance.js:121]: external link should have underline by default , it should be on hover.
and will you use this icon also with external link. We use for our external links
image

Comment (sudhanshutech) [reply src/sections/Projects/Sistent/components/link/guidance.js:121]: you will find in sistent

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

**Review by sudhanshutech — CHANGES_REQUESTED** [review]
@NishantSinghhhhh you can remove the topic download links , it doesn't make any sense because they same as basic links.

**Comment by l5io** [comment]
🚀 Preview for commit e11cdb4a86a2382811b932f8d5844c1f022c42d3 at: https://67173635c015755baff8bf8c--layer5.netlify.app

File: src/sections/Projects/Sistent/components/link/index.js (hunk 1 above)
Comment (NishantSinghhhhh) [reply src/sections/Projects/Sistent/components/link/index.js:77]: Sir I did that , but when no color was given , the link was rendering in black color and not green

File: src/sections/Projects/Sistent/components/link/guidance.js (hunk 3 above)
Comment (NishantSinghhhhh) [reply src/sections/Projects/Sistent/components/link/guidance.js:121]: Okay sir I will do this change

**Comment by l5io** [comment]
🚀 Preview for commit 63d5022acf74477228f381872ee27e03f3f6d044 at: https://671748c932848974d5d28df5--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 5685b9462026f6534b5882584806433bded1a1c0 at: https://671762903e850f867cc74570--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 0cf978c3c0e5a97d89dc32ccf3f2f4e854f5e495 at: https://671768518c4e2890167f2ba2--layer5.netlify.app

**Comment by NishantSinghhhhh** [comment]
@sudhanshutech , Sir I have done the changes you asked for

**Review by NishantSinghhhhh — COMMENTED** [review]
Updated link to Link

**Comment by l5io** [comment]
🚀 Preview for commit 862f881c2e8adb297c66cc0f0740ac09ad1a41e2 at: https://67178f868771dac5ab505ab9--layer5.netlify.app

**Comment by NishantSinghhhhh** [comment]
@sudhanshutech  sir ,How can i solve this dco issue ? , I am unable to find a way , should I raise one more PR for this issue or is there any way out

**Comment by NishantSinghhhhh** [comment]
@sudhanshutech sir can I raise 1 more PR , as There are lot many Commits that were not signed off by me , I will keep it in my mind to change the vs code settings to sign off the commit automatically

**Comment by l5io** [comment]
🚀 Preview for commit 8a4b8552586a07d32ea186fdf267744b96f09fa4 at: https://6717a2a150aba319f1ec9459--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 28a9d395d7a5326e7e6f4aa3437595146b89b90d at: https://6717a9e6fb455c2506a57fc9--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a284eee67254138cf7dc574d9d53e0f912087e96 at: https://6717e2f552118e7e27667c8b--layer5.netlify.app

**Comment by NishantSinghhhhh** [comment]
@sudhanshutech ,DCO issue is now removed

**Review by NishantSinghhhhh — COMMENTED** [review]
Done

**Review by NishantSinghhhhh — COMMENTED** [review]
Done.

**Comment by l5io** [comment]
🚀 Preview for commit eb4dbf80c8d172ed8ccfd466d6603c6983199490 at: https://6718b156b7a10f825732289b--layer5.netlify.app

**Comment by NishantSinghhhhh** [comment]
Are there any other changes required that makes my code suitable to merge it in the main branch

**Comment by l5io** [comment]
🚀 Preview for commit eb7ea29ce00e56b5ac45118f63cc42adcea0f185 at: https://67192c2db663b33fc5031454--layer5.netlify.app

**Comment by NishantSinghhhhh** [comment]
@nebula-aac , Sir I just resolved the conflicts , now it can be merged as it is free from conflicts

**Comment by l5io** [comment]
🚀 Preview for commit f5ebd918a352958f5ad867be1238a8bc95beb3c9 at: https://671a51bda22f2904dedec717--layer5.netlify.app

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: src/sections/Projects/Sistent/components/link/code-block.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React, { useState } from "react";
```
[the remark this answers is not in the document]
Comment (NishantSinghhhhh) [reply src/sections/Projects/Sistent/components/link/code-block.js:1]: Okay sir

**Comment by NishantSinghhhhh** [comment]
@nebula-aac @sudhanshutech , Sir I have deleted the code-block.js and used an already made component

**Comment by l5io** [comment]
🚀 Preview for commit b13dece31a1d097f3450f9a6ac1949d2369b0a90 at: https://671e132181a5f42297d489a5--layer5.netlify.app

**Review by sudhanshutech — APPROVED** [review]

**Pull request merged**